dagger/task.py

```diff
@@ -29,11 +43,35 @@ def get_all_dependencies(self):
             all_deps += dep.get_all_dependencies()
         return all_deps
 
+    def thread_safe_check_circular_dependencies(self, visiting_list):
```

Comment:
Please remove `thread_safe` from name - maybe move a hint to comments.

Also, please make it possible to call the method *without* a visiting_list parameter.

    (... visiting_list=None)

... and then in the body, set it to `[]` if it's None.